### PR TITLE
Use ancestor node finder for ruby expression statements

### PR DESCRIPTION
**Pull request by pokey**
Be sure to finish adding allowable parent types.  Note that saving triggered prettier formatter (which we will soon be enforcing), so to see the actual diff to your PR look at 9f11badf713d3ebde3e47e56d0f55dc24583cd59

## Checklist

- [x] I have added [tests](https://github.com/cursorless-dev/cursorless-vscode/blob/main/docs/contributing/test-case-recorder.md)

### CODE REVIEW

File: src/languages/ruby.ts [hunk 1]
```diff
@@ -24,41 +33,110 @@ const STATEMENT_TYPES = [
...
+  "symbol_array",
+  "true",
+  "unary",
+  "unless",
+  "until",
+  "while",
+  "yield",
+];
+
+// TODO: Add more allowable parent types
```
Comment (pokey) [inline src/languages/ruby.ts:98]: Make sure to fix this one on your PR

File: src/languages/ruby.ts [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
 } from "../util/nodeMatchers";
-import { NodeMatcherAlternative, ScopeType, SelectionWithEditor } from "../typings/Types";
+import {
+  NodeMatcherAlternative,
+  ScopeType,
+  SelectionWithEditor,
+} from "../typings/Types";
 import { SyntaxNode } from "web-tree-sitter";
 import { getNodeRange } from "../util/nodeSelectors";
+import { ancestorChainNodeFinder, patternFinder } from "../util/nodeFinders";
```
Comment (Will-Sommers) [inline src/languages/ruby.ts:18]: heyo - `ancestorChainNodeFinder` isn't referenced. I'll fix it in my PR.  Looking more at this, still trying to parse it. Will merge to move the Ruby PR forward.

File: src/languages/ruby.ts (hunk 1 above)
Comment (Will-Sommers) [reply src/languages/ruby.ts:98]: will do

File: src/languages/ruby.ts (hunk 2 above)
Comment (pokey) [reply src/languages/ruby.ts:18]: oh right yeah sorry I replaced it with `ancestorChainNodeMatcher` but then forgot to remove this one

**Pull request merged**